pkg/test/util/structpath/structpath.go

```diff
@@ -17,82 +17,105 @@ package structpath
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"reflect"
 	"regexp"
 	"strings"
 	"testing"
 
-	messagediff "gopkg.in/d4l3k/messagediff.v1"
-
 	"github.com/gogo/protobuf/jsonpb"
 	"github.com/gogo/protobuf/proto"
+
+	messagediff "gopkg.in/d4l3k/messagediff.v1"
+
 	"k8s.io/client-go/util/jsonpath"
 )
 
-var fixupNumericJSONComparison = regexp.MustCompile(`([=<>]+)\s*([0-9]+)\s*\)`)
-var fixupAttributeReference = regexp.MustCompile(`\[\s*'[^']+\s*'\s*]`)
+var (
+	fixupNumericJSONComparison = regexp.MustCompile(`([=<>]+)\s*([0-9]+)\s*\)`)
+	fixupAttributeReference    = regexp.MustCompile(`\[\s*'[^']+\s*'\s*]`)
+)
 
-type Structpath struct {
+type Instance struct {
 	structure interface{}
 	isJSON    bool
-	t         *testing.T
+}
+
+type InstanceForTest struct {
+	i *Instance
+	t *testing.T
+}
+
+func AssertProto(proto proto.Message) (*Instance, error) {
```

Comment:
not sure using the term 'assert' makes sense in this context any more

Reply:
Renamed.